### PR TITLE
Inline Fallback Functions For Debugging

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70656
* #70464
* #70871
* #70508
* __->__ #70463

Fix for https://github.com/pytorch/pytorch/issues/52940

When we call inlining on a fallback function, insert the runtime optimized version of its graph.

Differential Revision: [D33458651](https://our.internmc.facebook.com/intern/diff/D33458651)

### CODE REVIEW

File: torch/csrc/jit/passes/inliner.cpp [hunk 1]
```diff
@@ -30,8 +30,30 @@ void inlineCalls(Block* block) {
...
+              function_constant->hasAttribute(Symbol::attr("fallback"));
+          if (fallback && graphFunction->get_executor().isOptimized()) {
+            auto exec_plans =
+                graphFunction->get_executor().getDebugState().execution_plans;
+            if (exec_plans.size() != 0) {
+              g = exec_plans.begin()->second.graph;
+              // optimized_graph() calls Inline, so we only need to explicitly
+              // invoke inlining on the jit optimized graph with recursive
+              // fallback funciton calls
+              Inline(*g.get());
```
Comment (jjsjann123) [inline torch/csrc/jit/passes/inliner.cpp:48]: :+1:

**Review by jjsjann123 — APPROVED** [review]
Debugging improvements! Awesome!

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33458651).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33458651).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33458651).

**Pull request closed without merging**